### PR TITLE
Email and SMS services implementation

### PR DESCRIPTION
**Pull request by ruboarm**

### CODE REVIEW

**Comment by HaikAsatryan** [comment]
1. Twilio folder have email related models.
2. Please restore EmailChannels and SmsChannels previous version. This does not mean to delete current dictionaries. Just duplicate it for having strong typed versions. You can see in Ticketing application how it helps to create Email or Sms
3. Well done

File: src/Communicator/Services/Implementations/SmsService.cs [hunk 1]
```diff
@@ -1,18 +1,211 @@
...
 
+    private HttpClient GenerateProviderHttpClient()
+    {
+        var client = httpClientFactory.CreateClient(_channel);
+
+        var provider = GetSmsProvider(_smsConfiguration);
+
+        var properties = _smsConfiguration.Properties;
+
+        switch (provider)
```
Comment (Smbatyan) [inline src/Communicator/Services/Implementations/SmsService.cs:90]: create private method for switch case

File: src/Communicator/Services/Implementations/SmsService.cs [hunk 2]
```diff
@@ -1,18 +1,211 @@
...
+        {
+            case SmsProviders.Dexatel:
+                return await SendSmsViaDexatelAsync(smsMessage, cancellationToken);
+
+            case SmsProviders.Twilio:
+                return await SendSmsViaTwilioAsync(smsMessage, cancellationToken);
+
+            default:
+                throw new ArgumentOutOfRangeException();
```
Comment (Smbatyan) [inline src/Communicator/Services/Implementations/SmsService.cs]: change to InvalidOperationException

File: src/Communicator/Services/Implementations/SmsService.cs [hunk 3]
```diff
@@ -1,18 +1,211 @@
...
+            Data = new DexatelSmsSendRequestData
+            {
+                From = _smsConfiguration.From,
+                To = smsMessage.Recipients.MakeDistinct(),
+                Text = smsMessage.Message,
+                Channel = "SMS"
+            }
+        };
+
+        var response = await _httpClient.PostAsync(
```
Comment (Smbatyan) [inline src/Communicator/Services/Implementations/SmsService.cs]: separate method for _httpClient.PostAsync()

File: src/Communicator/Services/Implementations/SmsService.cs [hunk 4]
```diff
@@ -1,18 +1,211 @@
...
+    }
+
+    private async Task<List<GeneralSmsResponse>> SendSmsViaTwilioAsync(SmsMessage smsMessage,
+        CancellationToken cancellationToken)
+    {
+        var result = new List<TwilioSmsSendResponse>();
+
+        foreach (var phone in smsMessage.Recipients.MakeDistinct())
+        {
+            var response = await _httpClient.PostAsync(
```
Comment (Smbatyan) [inline src/Communicator/Services/Implementations/SmsService.cs]: same here
separate method

File: src/Communicator/Services/Implementations/SmsService.cs (hunk 1 above)
Comment (ruboarm) [reply src/Communicator/Services/Implementations/SmsService.cs:90]: done

File: src/Communicator/Services/Implementations/SmsService.cs (hunk 2 above)
Comment (ruboarm) [reply src/Communicator/Services/Implementations/SmsService.cs]: done

File: src/Communicator/Services/Implementations/SmsService.cs (hunk 3 above)
Comment (ruboarm) [reply src/Communicator/Services/Implementations/SmsService.cs]: done

File: src/Communicator/Services/Implementations/SmsService.cs (hunk 4 above)
Comment (ruboarm) [reply src/Communicator/Services/Implementations/SmsService.cs]: done

**Pull request merged**